servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConnectionContextProtocolTest.java

```diff
@@ -0,0 +1,118 @@
+
+import static io.servicetalk.http.api.HttpSerializationProviders.textDeserializer;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.http.netty.HttpProtocolConfigs.h1Default;
+import static io.servicetalk.http.netty.HttpProtocolConfigs.h2Default;
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static io.servicetalk.transport.netty.internal.AddressUtils.serverHostAndPort;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+@RunWith(Parameterized.class)
+public class ConnectionContextProtocolTest {
+
+    private enum Config {
+
+        HTTP_1_1(new HttpProtocolConfig[]{h1Default()}, false, HttpProtocolVersion.HTTP_1_1),
+        HTTP_2_0(new HttpProtocolConfig[]{h2Default()}, false, HttpProtocolVersion.HTTP_2_0),
+        SECURE_HTTP_1_1(new HttpProtocolConfig[]{h1Default()}, true, HttpProtocolVersion.HTTP_1_1),
+        SECURE_HTTP_2_0(new HttpProtocolConfig[]{h2Default()}, true, HttpProtocolVersion.HTTP_2_0),
+        ALPN_PREFER_HTTP_1_1(new HttpProtocolConfig[]{h1Default(), h2Default()}, true,
+                HttpProtocolVersion.HTTP_1_1),
+        ALPN_PREFER_HTTP_2_0(new HttpProtocolConfig[]{h2Default(), h1Default()}, true,
+                HttpProtocolVersion.HTTP_2_0);
+
+        final HttpProtocolConfig[] protocols;
+        final boolean secure;
+        final HttpProtocolVersion expectedProtocol;
+
+        Config(HttpProtocolConfig[] protocols, boolean secure, HttpProtocolVersion expectedProtocol) {
+            this.protocols = protocols;
+            this.secure = secure;
+            this.expectedProtocol = expectedProtocol;
+        }
+    }
+
+    @Rule
+    public final Timeout timeout = new ServiceTalkTestTimeout();
+
+    private final Config config;
+
+    public ConnectionContextProtocolTest(Config config) {
+        this.config = config;
+    }
+
+    @Parameters(name = "config={0}")
+    public static Object[] data() {
+        return Config.values();
+    }
+
+    @Test
+    public void testProtocol() throws Exception {
+        try (ServerContext serverContext = startServer(config);
+             BlockingHttpClient client = newClient(serverContext, config);
+             BlockingHttpConnection connection = client.reserveConnection(client.get("/"))) {
+
+            assertThat("Client-side connection protocol does not match expected value",
+                    connection.connectionContext().protocol(), equalTo(config.expectedProtocol));
+            assertThat("Server-side connection protocol does not match expected value",
+                    connection.request(connection.get("/")).payloadBody(textDeserializer()),
+                    equalTo(config.expectedProtocol.name()));
+        }
+    }
+
+    private static <T> ServerContext startServer(Config config) throws Exception {
+        final HttpServerBuilder builder = HttpServers.forAddress(localAddress(0))
```

Comment:
nit: static import here and below for `HttpClients.forSingleAddress`

Reply:
We do not use static imports for `HttpServers/HttpClients` in tests. I will keep it consistent with other test classes.